flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/io/benchmark/StreamNetworkBenchmarkEnvironment.java

```diff
@@ -238,18 +252,19 @@ private InputGate createInputGate(TaskManagerLocation senderLocation) throws Exc
 
 	private InputGateDeploymentDescriptor createInputGateDeploymentDescriptor(
 			TaskManagerLocation senderLocation,
-			int consumedSubpartitionIndex,
+			int gateIndex,
 			ResourceID localLocation) {
 
-		final ShuffleDescriptor[] channelDescriptors = Arrays.stream(partitionIds)
-			.map(partitionId ->
-				createShuffleDescriptor(localMode, partitionId, localLocation, senderLocation, consumedSubpartitionIndex))
-			.toArray(ShuffleDescriptor[]::new);
+		final ShuffleDescriptor[] channelDescriptors = new ShuffleDescriptor[channels];
+		for (int channelIndex = 0; channelIndex < channels; ++channelIndex) {
+			channelDescriptors[channelIndex] = createShuffleDescriptor(
+				localMode, partitionIds[gateIndex], localLocation, senderLocation, channelIndex);
```

Comment:
I am a bit torn for the `connectionIndex` value. If we use `channelIndex` as the value, it would create 1000 connections in theory if there are 1000 subpartitions in writer. I a bit prefer to use `gateIndex` instead to create less connections.